app/controllers/posts_controller.rb

```diff
@@ -11,4 +11,27 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = User.find(params[:user_id])
```

Comment:
- [ ] @AlyAtalla Great work implementing the changes requested, however you have not implemented the change requested, kindly ensure the author of the post is the current_user and not the user from the routes.